api/core/tools/provider/builtin/comfyui/tools/comfyui_stable_diffusion.py

```diff
@@ -0,0 +1,409 @@
+            lora_list.append(tool_parameters['lora_3'])
+            lora_strength_list.append(tool_parameters.get('lora_strength_3', 1))
+
+        print('invoke comfyui sd')
+
+        return self.text2img(base_url=base_url,
+                             model=model,
+                             prompt=prompt,
+                             negative_prompt=negative_prompt,
+                             width=width,
+                             height=height,
+                             steps=steps,
+                             sampler_name=sampler_name,
+                             scheduler=scheduler,
+                             cfg=cfg,
+                             lora_list=lora_list,
+                             lora_strength_list=lora_strength_list)
+
+    def get_checkpoints(self) -> list[str]:
+        """
+            get checkpoints
+        """
+        try:
+            base_url = self.runtime.credentials.get('base_url', None)
+            if not base_url:
+                return []
+            api_url = str(URL(base_url) / 'models' / 'checkpoints')
+            response = get(url=api_url, timeout=(2, 10))
+            if response.status_code != 200:
+                return []
+            else:
+                return response.json()
+        except Exception as e:
+            return []
+
+    def get_loras(self) -> list[str]:
+        """
+            get loras
+        """
+        try:
+            base_url = self.runtime.credentials.get('base_url', None)
+            if not base_url:
+                return []
+            api_url = str(URL(base_url) / 'models' / 'loras')
+            response = get(url=api_url, timeout=(2, 10))
+            if response.status_code != 200:
+                return []
+            else:
+                return response.json()
+        except Exception as e:
+            return []
+
+    def get_sample_methods(self) -> list[str]:
+        """
+            get sample method
+        """
+        try:
+            base_url = self.runtime.credentials.get('base_url', None)
+            if not base_url:
+                return []
+            api_url = str(URL(base_url) / 'object_info' / 'KSampler')
+            response = get(url=api_url, timeout=(2, 10))
+            if response.status_code != 200:
+                return []
+            else:
+                return response.json()['KSampler']['input']['required']['sampler_name'][0], response.json()['KSampler']['input']['required']['scheduler'][0]
+        except Exception as e:
+            return [], []
+
+    def validate_models(self) -> Union[ToolInvokeMessage, list[ToolInvokeMessage]]:
+        """
+            validate models
+        """
+        try:
+            base_url = self.runtime.credentials.get('base_url', None)
+            if not base_url:
+                raise ToolProviderCredentialValidationError('Please input base_url')
+            model = self.runtime.credentials.get('model', None)
+            if not model:
```

Comment:
<img width="307" alt="image" src="https://github.com/user-attachments/assets/5932bdb5-4e11-43e2-a1e7-9e10c8ce92e3">
It seems this input is missing?

Reply:
You should input `http://127.0.0.1:8188`